lib/models/main_survey/main_question.dart

```diff
@@ -0,0 +1,28 @@
+import 'package:json_annotation/json_annotation.dart';
+
+part 'main_question.g.dart';
+
+@JsonSerializable()
+class MainQuestion {
+  final String text;
+  bool value;
+
+  MainQuestion({required this.text, required this.value});
+
+  factory MainQuestion.fromJson(Map<String, dynamic> json) =>
+      _$MainQuestionFromJson(json);
+
+  Map<String, dynamic> toJson() => _$MainQuestionToJson(this);
+
+  MainQuestion createToggle() {
+    return MainQuestion(text: text, value: value);
+  }
+
```

Comment:
Questo metodo non viene usato da nessuna parte. 
Lo eliminerei oppure lo sostituirei con questo e lo utilizzerei nell' `on<EventToggleQuestion>`  del bloc.

```suggestion
    MainQuestion toggle() {
        return MainQuestion(text: text, value: !value);
    }
```

utilizzandolo così:

```dart
on<EventToggleMainQuestion>((event, emit) {
    List<MainQuestion> newState = List.from(state.mainQuestions);
    newState[event.index] = newState[event.index].toggle();
    emit(MainSurveyState(newState));
});
```